src/test/integration/java/com/commercetools/payment/BasePaymentIT.java

```diff
@@ -124,9 +139,49 @@ protected String createCartAndPayment(@Nonnull SphereClient sphereClient) {
         CartDraft dummyComplexCartWithDiscounts = CartDraftBuilder.of(getDummyComplexCartDraftWithDiscounts())
                 .currency(EUR)
                 .build();
-        return sphereClient.execute(CartCreateCommand.of(dummyComplexCartWithDiscounts));
+        CartDraft cartDraft = getProductsInjectedCartDraft(sphereClient, dummyComplexCartWithDiscounts);
+
+        return sphereClient.execute(CartCreateCommand.of(cartDraft));
+    }
+
+    public static CartDraft getProductsInjectedCartDraft(@Nonnull SphereClient sphereClient, CartDraft dummyComplexCartWithDiscounts) {
```

Comment:
I wonder why you would add here custom code instead of extending existing [JSON](https://github.com/commercetools/commercetools-paypal-plus-integration/blob/master/src/test/integration/resources/mockData/paymentHandler/dummyCartDraftWithDiscounts.json) with required data which is used by `getDummyComplexCartDraftWithDiscounts()`?

Reply:
It is because this particular method is being called for multiple test suites and we should have to have unique test product type key in each scenario. https://github.com/commercetools/commercetools-paypal-plus-integration/pull/142/files/3c474415dc1c52d4e1bb3b712b80aba5b7b257d9#diff-1218b3d5cbb34b36dcab78c88e41b01bR149

Reply:
So why not have unique product key for all tests?

Reply:
There are different test cases which uses the above `getDummyComplexCartDraftWithDiscounts` method like here:https://github.com/commercetools/commercetools-paypal-plus-integration/blob/aa0be5dbd8c5d1724ca641ba62582daa4bef1bde/src/test/integration/java/com/commercetools/payment/BasePaymentIT.java#L124
So, we cannot guarantee the test case execution order. I think it's better to create a random tax category and remove it once the test case has finished. It makes the test case execution independent of others.